app/src/main/java/com/google/android/gms/samples/vision/ocrreader/aadhaar/HomeActivity.java

```diff
@@ -0,0 +1,325 @@
+
+    // UI Elements
+    TextView tv_sd_uid,tv_sd_name,tv_sd_gender,tv_sd_yob,tv_sd_co,tv_sd_vtc,tv_sd_po,tv_sd_dist,
+            tv_sd_state,tv_sd_pc,tv_cancel_action;
+    LinearLayout ll_scanned_data_wrapper,ll_data_wrapper,ll_action_button_wrapper, ll_scan_qr_wrapper;
+
+    private static final int RC_HANDLE_CAMERA_PERM = 2;
+
+
+    // Storage
+    Storage storage;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        //hide the default action bar
+        getSupportActionBar().hide();
+        setContentView(R.layout.activity_home);
+
+        // init the UI Elements
+        tv_sd_uid = (TextView)findViewById(R.id.tv_sd_uid);
+        tv_sd_name = (TextView)findViewById(R.id.tv_sd_name);
+        tv_sd_gender = (TextView)findViewById(R.id.tv_sd_gender);
+        tv_sd_yob = (TextView)findViewById(R.id.tv_sd_yob);
+        tv_sd_co = (TextView)findViewById(R.id.tv_sd_co);
+        tv_sd_vtc = (TextView)findViewById(R.id.tv_sd_vtc);
+        tv_sd_po = (TextView)findViewById(R.id.tv_sd_po);
+        tv_sd_dist = (TextView)findViewById(R.id.tv_sd_dist);
+        tv_sd_state = (TextView)findViewById(R.id.tv_sd_state);
+        tv_sd_pc = (TextView)findViewById(R.id.tv_sd_pc);
+        tv_cancel_action = (TextView)findViewById(R.id.tv_cancel_action);
+
+        ll_scanned_data_wrapper = (LinearLayout)findViewById(R.id.ll_scanned_data_wrapper);
+        ll_data_wrapper = (LinearLayout)findViewById(R.id.ll_data_wrapper);
+        ll_action_button_wrapper = (LinearLayout)findViewById(R.id.ll_action_button_wrapper);
+        ll_scan_qr_wrapper = (LinearLayout) findViewById(R.id.ll_scan_qr_wrapper);
+
+        final int rc = ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA);
+
+        ll_scan_qr_wrapper.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                scanNow(v);
+
+            }
+        });
+
+        //init storage
+        storage = new Storage(this);
+    }
+
+
+
+    /**
+     * onclick handler for scan new card
+     * @param view
+     */
+    public void scanNow( View view){
+        IntentIntegrator integrator = new IntentIntegrator(this);
+        integrator.setDesiredBarcodeFormats(IntentIntegrator.QR_CODE_TYPES);
+        integrator.setPrompt("Scan a Aadharcard QR Code");
+        integrator.setResultDisplayDuration(500);
+        integrator.setCameraId(0);  // Use a specific camera of the device
+        integrator.initiateScan();
+    }
+
+    /**
+     * function handle scan result
+     * @param requestCode
+     * @param resultCode
+     * @param intent
+     */
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        //retrieve scan result
+        IntentResult scanningResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, intent);
```

Comment:
The issue with this is the UI and the scanner itself is out of our control. In Yoti we would want to customise the UI to our liking.